python/genhisto.py

```diff
@@ -0,0 +1,78 @@
+import argparse
+import subprocess
+import numpy as np
+
+def main():
+
+    parser = argparse.ArgumentParser(description="Generate histogram of nifti-1 image")
+    group = parser.add_mutually_exclusive_group(required=False)
+    group.add_argument("-nb", "--num_bins", help="The number of bins in your histogram (integer value). ", type=int)
+    group.add_argument("-br", "--bin_range", help="The interval range of the bins (float)", type=float)
+    parser.add_argument("--min_val", help="The minimum voxel value of the image", type=float)
+    parser.add_argument("--max_val", help="The maximum voxel value of the image", type=float)
+    
+
+    parser.add_argument("--num_reducers", help="The number of reducers to be used", type=int)
+    parser.add_argument("--num_mappers", help="The number of mappers to be used", type=int)
+
+    parser.add_argument("-o", "--output", help="The output folder", type=str, default="HistogramOutput")
+    
+    required = parser.add_argument_group('required arguments')
+    required.add_argument("-i", "--input_file", help="The input file", type=str, required=True)
+    required.add_argument("-hj", "--hadoop_streaming_jar", help="The hadoop streaming jar", required=True)
+
+    args = parser.parse_args()
+    
+    number_bins = args.num_bins
+    bin_range = args.bin_range
+
+    data_min = args.min_val
+    data_max = args.max_val
+
+    num_reducers = args.num_reducers
+    num_mappers = args.num_mappers
+
+    input_file = args.input_file
+    output_folder = args.output
+
+    hadoop_jar = args.hadoop_streaming_jar
+
+    # will get the min/max value of the image if min/max is not already provided
+    if(data_min == None or data_max == None):
+        find_min_max_command = 'hadoop jar {0} -D mapreduce.job.reduces=0 -D mapreduce.job.map={1} -input {2} -output {3} -mapper "python minmaxmap.py" -file minmaxmap.py'.format(hadoop_jar, num_mappers, input_file, output_folder + '_temp')
+        #run bash command
+        subprocess.call(find_min_max_command, shell=True)
+
+        get_minmax_out = None
+
+        #save minMax result to intermediate folder
+        #should perhaps delete folder after result is extracted..
+        if 'file:///' not in output_folder:
+            get_minmax_out = 'hdfs dfs -cat {0}/* | sort'.format(output_folder + '_temp')
+        else:
+            get_minmax_out = 'cat {0}/* | sort'.format(output_folder.replace('file:///', '') + '_temp')
```

Comment:
That '_temp' should be in a variable (it's used on line 42 already). To avoid issues with that, use `tempfile.mkdtemp()`.